### ISSUE TITLE
Feature req: Expo support for React Native library

### ISSUE THREAD

**Issue by paolodamico** [PostHog/posthog#1174]
**Is your feature request related to a problem? The [React Native library](https://github.com/PostHog/posthog-react-native) cannot run on [Expo's stand-alone apps](https://docs.expo.io/distribution/building-standalone-apps/) because as mentioned [here](https://docs.expo.io/introduction/managed-vs-bare/#workflow-comparison), managed apps cannot link native dependencies.

**Describe the solution you'd like** Direct integration via the Expo SDK.

**Describe alternatives you've considered** Doing a manual integration via the API, but it would take too much time and most of the benefits of the capturing library would be lost.

**Additional context** Building with Expo is a very convenient way to deploy native apps, I would imagine a number of apps run on their managed workflow.

***Thank*** you for your feature request - we love each and every one!

**Comment by timgl** [comment]
Thanks for creating this issue! Expo support would be awesome

@mariusandra Any idea what would be required to do this?

**Comment by mariusandra** [comment]
Hi @timgl - We should totally do this, yet it's a non-trivial amount of work to get a RN library that's at parity with the current native client libraries with regards to autocapture.

The native libraries provide us with a stream of tap and screenview events, which we won't be able to get with RN alone.

Heap had to do [quite a bit of innovative engineering](https://heap.io/blog/engineering/how-we-leveraged-asts-and-babel-to-capture-everything-on-react-native-apps) to add this support. We could do something similar though. I'm pretty comfortable with AST trees by now :).

Alternatively, it should be possible to use the existing [posthog-node](https://github.com/PostHog/posthog-node) library to manually capture events... and with a bit of tinkering the upcoming [posthog-js-lite](https://github.com/PostHog/posthog-js-lite) library should also work. However those will not give the autocapture or feature flag options.

**Comment by Immortalin** [comment]
https://docs.expo.io/versions/latest/sdk/segment/

@timgl

**Comment by adjwills** [comment]
Hi just wanted to add a vote on this req. Would be really useful for us to have Expo integration, trying the posthog-node lib on expo but obviously this misses some really great features like auto-capture.

Also FYI, papercups got a fair few requests for expo integration https://news.ycombinator.com/item?id=24226979 and prioritised it pretty quick https://news.ycombinator.com/item?id=24226979. Expo is attractive for smaller teams wanting to avoid codebase fragmentation. On the assumption that posthog appeals to a similar base it would be a win.

**Comment by mariusandra** [comment]
Hey @adjwills , thanks for the vote! I think we don't have enough bandwidth to squeeze this into the current sprint/release, as it's quite a bit of work to get done, especially for autocapture (see what [heap had to do](https://heap.io/blog/engineering/how-we-leveraged-asts-and-babel-to-capture-everything-on-react-native-apps) to get this working), but we'll definitely take note of this!

**Comment by adjwills** [comment]
Hey @mariusandra , thanks for the response! Totally understand that this is a fair bit of work, it will certainly be really useful to us if/when it's available. 

For now we'll look at a more minimal implementation with the existing posthog-node library, that looks like the best route to me. Assuming that works well we'd be keen to capture more value from the autocapture features etc. (I've tested all those with a react-js web build and am super impressed with what you guys have built!)

**Comment by paolodamico** [comment]
+1 from [user request](https://posthog.slack.com/archives/C01GYST86H5/p1612261208006200).

**Comment by anandvc** [comment]
+1 for Expo support with autocapture/session-recordings from my team's side as well.

**Comment by keithschacht** [comment]
+1 for Expo support

**Comment by Chapnik13** [comment]
+1 For Expo support

**Comment by tachyon-ops** [comment]
+1 for Expo support

**Comment by dqii** [comment]
+1 for Expo support

**Comment by Eric-Hei** [comment]
+1 for Expo support

**Comment by paolodamico** [comment]
Looks like we have some interest thing @yakkomajuri

**Comment by keithschacht** [comment]
FYI for others: we got it working in Expo for the time being by creating two of our own server endpoint (in ruby). Our expo client hits our endpoint to get a distinct key, the front-end saves this. Then each time we want to log an event, we hit our own API endpoint. Our server uses the Ruby PostHog client and creates a PostHog event using this server-side system.

It was about 20 lines of code total. Not the most elegant but it got things working!

**Comment by dakshshah96** [comment]
+1 for Expo support

**Comment by jokull** [comment]
Expo support would be huge.

**Comment by TijsM** [comment]
+1 for Expo support

**Comment by kovilsingh** [comment]
@paolodamico would we have this anytime soon?

+1 for expo support!

**Comment by mariusandra** [comment]
YES! We're actively working on it: https://github.com/PostHog/posthog-js-lite/pull/4

CC @benjackwhite

**Comment by benjackwhite** [comment]
Ooh I didn't see this issue 😅 I am indeed hot on the heels of this issue and will be looking for testers shortly. 

Stay tuned... I'll post here once we have something on npm for testing 💪

**Comment by benjackwhite** [comment]
For anybody who is subscribed to this issue - it's here! `posthog-react-native@2.0.0` is out and is 100% pure React Native, utilizing only standard expo libraries to get access to the bits of native functionality it needs to do its job. 
It has all the important stuff it used to have as well as...
* Better feature flags implementation including [authentication persistence](https://posthog.com/docs/user-guides/feature-flags#persisting-feature-flags-across-authentication-steps) and helpful hooks
* Much better Screen tracking hooking into `@react-native-navigation/native`
* Autocapture for touches
* Support for managed Expo projects (no ejecting!)

Now that this is fully JS (well, Typescript) based it is easier than ever for us to improve things and iterate so speak up if there's something missing!

[Check out the docs for more info](https://posthog.com/docs/integrate/client/react-native)

🙌

**Comment by paolodamico** [comment]
Amazing news @benjackwhite! Guess we can close this now 🎊
